sympy/solvers/diophantine.py

```diff
@@ -678,7 +623,7 @@ def divisible(a, b):
     """
     Returns `True` if ``a`` is divisible by ``b`` and `False` otherwise.
     """
-    return igcd(int(a), int(b)) == abs(int(b))
+    return not a % b
```

Comment:
the Diophantine routines should be careful to always work in integers, and to make only the public function return SymPy types. Right now, however, too many functions are public, I think.